smart_open/smart_open_lib.py

```diff
@@ -269,6 +276,34 @@ def __exit__(self, type, value, traceback):
         self.read_key.close()
 
 
+class SSHOpenRead(object):
+    """
+    Implement streamed reader from remote server using SSH
+
+    """
+    def __init__(self, parsed_uri):
+        if parsed_uri.scheme not in ("ssh"):
+            raise TypeError("can only process remote files via ssh")
+        self.parsed_uri = parsed_uri
+
+    def __iter__(self):
+        ssh = subprocess.Popen("ssh -o StrictHostKeyChecking=no " + self.parsed_uri.netloc + " 'cat " + self.parsed_uri.uri_path + "'",
+            stdout=subprocess.PIPE, shell=True)
```

Comment:
Choices on this line introduce risks that, while perhaps justified in many/most usage situations, should be clearly documented and perhaps require a user's explicit opt-in. (1) Users should know (perhaps by choosing a non-default option) that there's no host-key-checking; (2) composing a locally- and remotely- shelled execution string from the URI parts could introduce arbitrary execution risks at both endpoints, so extra care must be taken where such paths originate.